odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1961,48 +1933,12 @@ def get_view_xmlid(self):
 
     @api.model
     def render_public_asset(self, template, values=None):
-        template = self.sudo().browse(self.get_view_id(template))
-        template._check_view_access()
-        return template.sudo()._render(values, engine="ir.qweb")
+        view = self.sudo().browse(self.get_view_id(template))
+        view._check_view_access()
+        return self.sudo()._render_template(template, values=values)
```

Comment:
```suggestion
        return self.['ir.qweb']._render(template, values=values)
```
does not work here?

Reply:
If this will work, unless there is a public template somewhere that t-calls a private template. But I don't think that currently exists.
I will change it.